code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm

```diff
@@ -43,17 +43,26 @@
 	var/datum/gas_mixture/remove_output = air_output.remove_ratio(0.9)
 
 	var/coolant_temperature_delta = remove_input.temperature - remove_output.temperature
+	var/power_usage
+	var/input_capacity = remove_input.heat_capacity()
+	var/output_capacity = remove_output.heat_capacity()
+	var/thermal_equilibrium
 
-	if(coolant_temperature_delta > 0)
-		var/input_capacity = remove_input.heat_capacity()
-		var/output_capacity = remove_output.heat_capacity()
+	//Exchange heat between the nodes.
+	if(coolant_temperature_delta)
+		thermal_equilibrium = (input_capacity * remove_input.temperature + output_capacity * remove_output.temperature) / (input_capacity + output_capacity)
+		remove_input.temperature = max(thermal_equilibrium, TCMB)
+		remove_output.temperature = max(thermal_equilibrium, TCMB)
+	else
+		thermal_equilibrium = remove_input.temperature
 
-		var/cooling_heat_amount = (heat_transfer_rate * 0.01) * CALCULATE_CONDUCTION_ENERGY(coolant_temperature_delta, output_capacity, input_capacity)
-		remove_output.temperature = max(remove_output.temperature + (cooling_heat_amount / output_capacity), TCMB)
-		remove_input.temperature = max(remove_input.temperature - (cooling_heat_amount / input_capacity), TCMB)
-		update_parents()
+	//Pump heat against the gradient.
+	var/heat_amount = CALCULATE_CONDUCTION_ENERGY((thermal_equilibrium - max(thermal_equilibrium * 0.15, TCMB)) * heat_transfer_rate * 0.01, input_capacity, output_capacity)
```

Comment:
why the double () wrap? should be done on the define's end

Reply:
So that the energy that gets transferred gets multiplied by the heat transfer rate, rather than the fake temperature delta or whatever.